packages/create/src/generators/app-haunted/templates/_MyApp.js

```diff
@@ -0,0 +1,128 @@
+import { html } from 'lit-html';
+import { component, useState } from 'haunted';
+import { classMap } from 'lit-html/directives/class-map.js';
+import { openWcLogo } from './open-wc-logo.js';
+
+import '../../page-main/page-main.js';
+import '../../page-one/page-one.js';
+import { templateAbout } from './templateAbout.js';
+
+function addActiveIf(el, page) {
+  return classMap({ active: el.page === page });
+}
+
+function renderPage(page) {
+  switch (page) {
+    case 'main':
+      return html`
+        <page-main .logo=${openWcLogo}></page-main>
+      `;
+    case 'pageOne':
+      return html`
+        <page-one></page-one>
+      `;
+    case 'about':
+      return templateAbout;
+    default:
+      return html`
+        <p>Page not found try going to <a href="#main">Main</a></p>
+      `;
+  }
+}
+
+function styles() {
+  return html`
+    <style>
+      :host {
+        min-height: 100vh;
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        justify-content: flex-start;
+        font-size: calc(10px + 2vmin);
+        color: #1a2b42;
+        margin: 0 auto;
+      }
+      :host > header,
+      :host > main,
+      :host > p {
+        display: flex;
+      }
+      header {
+        width: 100%;
+        background: #fff;
+        border-bottom: 1px solid #ccc;
+      }
+      header ul {
+        display: flex;
+        justify-content: space-between;
+        min-width: 400px;
+        margin: 0 auto;
+        padding: 0;
+      }
+      header ul li {
+        display: flex;
+      }
+      header ul li a {
+        color: #ccc;
+        text-decoration: none;
+        font-size: 18px;
+        line-height: 36px;
+      }
+      header ul li a:hover,
+      header ul li a.active {
+        color: #000;
+      }
+      main {
+        flex-grow: 1;
+      }
+      .app-footer {
+        color: #a8a8a8;
+        font-size: calc(12px + 0.5vmin);
+        align-items: center;
+      }
+      .app-footer a {
+        margin-left: 5px;
+      }
+    </style>
+  `;
+}
+
+const <%= className %>Element = el => {
+  const [page, setPage] = useState(el.page || 'main');
+
+  function clickPageLink(ev) {
+    ev.preventDefault();
+    setPage(ev.target.hash.substring(1));
+  }
+
+  return html`
+    ${styles()}
+    <header>
+      <ul>
+        <li>
+          <a href="#main" class=${addActiveIf(el, 'main')} @click=${clickPageLink}>Main</a>
```

Comment:
i'm curious, what will lit-html bind this to? what will I get if I try to access `this` in the handler? Is it more idiomatic haunted to curry in `el`?

Reply:
This was written by matthew himself in https://github.com/matthewp/my-app/blob/master/packages/my-app/src/MyApp.js#L114 so im inclined to go with it :p Although `this` should also work, see:

```js
/**
 * function FireEventsChild({myProp}) {
 *  return html`
 *    <button @click=${() => dispatch(this, 'event-fired', myProp)}>
 *      clickity
 *    </button>
 *  `;
 * }
 */
```